### PR TITLE
SE - Inject PageConfigurationHelper into PageAdapter

### PR DESCRIPTION
**Pull request by jcalcines**

### CODE REVIEW

File: site/components/src/main/java/com/visitscotland/brxm/components/content/service/AuthorService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                }
+
+                return Optional.of(blog);
+            }
+        } catch (PageCompositionException e) {
+            helper.getPageTemplate().addErrorMessage("The blog section threw an error and cannot be included in this page");
+            logger.warn(e.getMessage(), e);
+        }
+        return Optional.empty();
+    }
```
[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/main/java/com/visitscotland/brxm/components/content/service/AuthorService.java:64]: Good Catch I don't know how you figured that out but, It is a known issue, we are temporarly duplicating the data so we don't create breaking changes. The removal is already scheduled by a the ticket VS-1556.

[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/main/java/com/visitscotland/brxm/components/content/service/AuthorService.java:64]: VS-1556 is enough for tracking

File: site/components/src/main/java/com/visitscotland/brxm/pagebuilder/page/ItineraryPageAdapter.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class ItineraryPageAdapter implements PageAdapter<Itinerary> {
+
+    private final Logger log = LoggerFactory.getLogger(ItineraryPageAdapter.class);
+
+    @Override
+    public PageIntro getPageIntro(PageCompositionHelper pageConfig) {
+        log.warn("Page intro composition for Itinerary is not implemented yet.");
+
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/main/java/com/visitscotland/brxm/pagebuilder/page/ItineraryPageAdapter.java]: That will be fixed in the follow up job

File: site/components/src/main/java/com/visitscotland/brxm/pagebuilder/PageCompositionHelper.java [hunk 3]
```diff
@@ -30,12 +38,65 @@ public Locale getLocale(){
...
+        }
+
+        //TODO: pageIntro might be replaced by pageTemplate
+        request.setModel("pageIntro", template);
+        request.setModel(PAGE_TEMPLATE, template);
+
+        this.pageTemplate = template;
+
+        return template;
+    }
```
[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/main/java/com/visitscotland/brxm/pagebuilder/PageCompositionHelper.java:91]: What if I was expecting an edge case (an specific configuration for a page compound) that can be use all the logic from this class but this method?

This null check tried to capture a misuse of that edge case

File: site/components/src/test/java/com/visitscotland/brxm/components/content/service/PageLabelsTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        verify(pageConfig, never()).addAllSiteLabels(PageLabels.CMS_MESSAGES_BUNDLE);
+    }
+
+    @Test
+    @DisplayName("Should register global labels")
+    void includeGeneralLabels_verifies_all_global_label_registrations() {
+        pageLabels.includeGeneralLabels(pageConfig, false);
+
+        verify(pageConfig, atLeast(2)).addGlobalLabel(anyString());
+    }
```
[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/test/java/com/visitscotland/brxm/components/content/service/PageLabelsTest.java:53]: 1. The purpose of the test is to validate that a set of labels is added, the labels themselves are not an important detail and they might vary on time. What needs to be validated is that the labels several "label files" are injected into the payload
2. good shout

File: site/components/src/test/java/com/visitscotland/brxm/pagebuilder/page/GeneralPageAdapterTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        when(page.getCategoryLinks()).thenReturn(mock(PageLinksSectionCompound.class));
+        when(categoryCardsMapper.getCategoryCards(eq(LOCALE), any()))
+                .thenReturn(new LinksModule<>());
+
+
+        PageIntro result = adapter.getPageIntro(pageConfig);
+
+        assertSame(page, result.getHippoBean());
+        assertNotNull(result.getCategorySection());
+    }
```
[the remark this answers is not in the document]
Comment (jcalcines) [reply site/components/src/test/java/com/visitscotland/brxm/pagebuilder/page/GeneralPageAdapterTest.java:57]: What do you mean? how would you fix it?

**Review by jlrhurta — APPROVED** [review]
